test/geos_capi/polygon_test.rb

```diff
@@ -109,7 +109,17 @@ def test_buffer_with_style
     buffered_line_string =
       line_string.buffer_with_style(0.3, RGeo::Geos::CAP_SQUARE, RGeo::Geos::JOIN_MITRE, 5)
 
-    assert_equal polygon, buffered_line_string
+    # having issues with floating point errors on some systems
+    # 4.3 -> 4.29999999999999, for example, and throws an error
+    # iterating through points and using assert_in_delta instead
+    # of assert_equal
+    b_coords = buffered_line_string.exterior_ring.coordinates
+    p_coords = polygon.exterior_ring.coordinates
+    p_coords.zip(b_coords).each do |p_coord, b_coord|
+      p_coord.zip(b_coord).each do |pt1, pt2|
+        assert_in_delta(pt1, pt2, 1e-7)
+      end
+    end
```

Comment:
This one is not satisfying, but this PR is not the place to fix floating point errors!